src/index.tsx

```diff
@@ -218,15 +233,64 @@ class Cropper extends React.Component<CropperProps, State> {
 
   computeSizes = () => {
     const mediaRef = this.imageRef || this.videoRef
+
     if (mediaRef && this.containerRef) {
       this.containerRect = this.containerRef.getBoundingClientRect()
+      const containerAspect = this.containerRect.width / this.containerRect.height
+      const naturalWidth = this.imageRef?.naturalWidth || this.videoRef?.videoWidth || 0
+      const naturalHeight = this.imageRef?.naturalHeight || this.videoRef?.videoHeight || 0
+      const isMediaScaledDown = mediaRef.width < naturalWidth || mediaRef.height < naturalHeight
+      const mediaAspect = naturalWidth / naturalHeight
+
+      // We do not rely on the offsetWidth/offsetHeight if the media is scaled down
+      // as the values they report are rounded. That will result in precision losses
+      // when calculating zoom. We use the fact that the media is positionned relative
+      // to the container. That allows us to use the container's dimensions
+      // and natural aspect ratio of the media to calculate accurate media size.
+      // However, for this to work, the container should not be rotated
+      let renderedMediaSize: Size
+
+      if (isMediaScaledDown) {
+        switch (this.props.objectFit) {
+          default:
+          case 'contain':
+            renderedMediaSize =
+              containerAspect > mediaAspect
+                ? {
+                    width: this.containerRect.height * mediaAspect,
+                    height: this.containerRect.height,
+                  }
+                : {
+                    width: this.containerRect.width,
+                    height: this.containerRect.width / mediaAspect,
+                  }
+            break
+          case 'horizontal-cover':
```

Comment:
If you try this version on the demo sandbox for `objectFit` (https://codesandbox.io/s/react-easy-crop-forked-1mzi1), something seems wrong for `horizontal-cover` and `vertical-cover`.

Here's the original version: https://codesandbox.io/s/react-easy-crop-forked-p9r34

Reply:
You used the older version of the build, before `Support objectFit prop` commit, here's the same codesandbox with the latest version - https://codesandbox.io/s/react-easy-crop-forked-n6kwy